### PR TITLE
Remove text opacity CSS variables from `::marker`

### LINKED ISSUE #8610: Safari ignores alpha channel in colors in `::marker`, which makes styling them with Tailwind impossible
**What version of Tailwind CSS are you using?**

v3.1.2

**What build tool (or framework if it abstracts the build tool) are you using?**

<details>
<summary>`postcss`</summary>

```js
module.exports = {
  plugins: {
    "postcss-import": {},
    "tailwindcss": {},
    "autoprefixer": {},
  }
}
```
</details>

**What version of Node.js are you using?**

v12.22.9

**What browser are you using?**

Safari 15.5 on macOS 12.4

**Reproduction URL**

https://play.tailwindcss.com/iea3Szbz1t

**Describe your issue**

On Safari, all of the following examples work correctly:

```css
::marker {
  color: red;
}
::marker {
  color: #f00;
}
::marker {
  color: rgb(255 0 0);
}
```

But these are completely ignored:

```css
::marker {
  color: rgba(255 0 0 50);
}
::marker {
  color: rgb(255 0 0 / 50);
}
```

I discovered this while playing with arbitrary variants, trying to lighten the ordinals in an ordered list. When that didn't work, I thought, "that's fine, I can just use an arbitrary value!" but turns out that `[&::marker]:text-[#ff0000]` gets translated to

```css
.\[\&\:\:marker\]\:text-\[\#ff0000\]::marker {
  --tw-text-opacity: 1;
  color: rgb(255 0 0 / var(--tw-text-opacity));
}
```

Which makes it impossible to use Tailwind to style marker colors.

I realize this is more of a Safari bug than a Tailwind bug, but I'm curious as to why you're translating a hardcoded `[red]` into an RGBA value, and whether that's something that could be addressed to allow styling list markers with Tailwind 🤔

### PR DESCRIPTION
**Pull request by thecrypticace**
The rules that use the `::marker` pseudo class only allow certain variables to be defined. In Safari it's likely there's an allowlist of some kind and it doesn't include custom properties. This prevents the default Tailwind text colors from working because we reference alpha values as the textOpcity plugin is on by default. To work around this you can disable the textOpacity plugin. This is definitely a browser bug that needs to be reported however it's also super not ideal that the out-of-the-box experience is broken for Tailwind here so we'll do something similar to our solution with `::visited` and strip the text opacity property from rules and declarations.

This additionally fixes a bug where mutations to `container` and use of `modifySelector` did not preserve selectors when returning an array from a variant function.

Basically it turns this:
```css
.marker\:text-red-500 *::marker {
  --tw-text-opacity: 1;
  color: rgb(239 68 68 / var(--tw-text-opacity));
}
.marker\:text-red-500::marker {
  --tw-text-opacity: 1;
  color: rgb(239 68 68 / var(--tw-text-opacity));
}
```

into this:
```css
.marker\:text-red-500 *::marker {
  color: rgb(239 68 68);
}
.marker\:text-red-500::marker {
  color: rgb(239 68 68);
}
```

Fixes #8610

### CODE REVIEW

**Review by RobinMalfait — COMMENTED** [review]
A few questions but I think that makes sense. Can't wait to remove the `modifySelectors` code in v4 so that we only keep the new API and the container mutations directly 😄

File: tests/variants.test.css [hunk 1]
```diff
@@ -111,17 +111,15 @@
...
-  --tw-text-opacity: 1;
-  color: rgb(239 68 68 / var(--tw-text-opacity));
-}
 .marker\:text-lg::marker {
   font-size: 1.125rem;
   line-height: 1.75rem;
 }
+.marker\:text-red-500 *::marker {
+  color: rgb(239 68 68);
+}
```
[the remark this answers is not in the document]
Comment (adamwathan) [reply tests/variants.test.css:120]: I think what's happening is the utilities are being grouped by utility now instead of by the selector, it's like:

```
.marker\:poop *::marker {}
.marker\:poop::marker {}

.marker\:banana *::marker {}
.marker\:banana::marker {}

.marker\:tonsils *::marker {}
.marker\:tonsils::marker {}
```

...instead of:

```
.marker\:poop *::marker {}
.marker\:banana *::marker {}
.marker\:tonsils *::marker {}

.marker\:poop::marker {}
.marker\:banana::marker {}
.marker\:tonsils::marker {}
```

I'd have to think about it harder but it feels like there was an important reason we originally did it by selector, so we probably want to be careful here. It's tricky admittedly for the reasons we talked about the other week, where variants registered within a function share a sort order or whatever.

I wonder if what we really want is to make this API possible for this one:

```
addVariant('marker', [() => {...}, () => {...}])
```

...so we can know the count up front and adjust the sort appropriately?

[the remark this answers is not in the document]
Comment (thecrypticace) [reply tests/variants.test.css]: Good news @adamwathan that API already works :D — I've updated the PR to switch to it

**Comment by thecrypticace** [comment]
@RobinMalfait @adamwathan I've addressed your feedback — ping me if there's anything else that needs to be looked at. I'd like to merge this in and tag a release today.

**Review by RobinMalfait — APPROVED** [review]
I think this looks good now!
